### PR TITLE
Tutorial notebook for coadd catalog with GCR access 

### PR DESCRIPTION
**Pull request by EiffL**
I'm opening this PR to pin this here, and have a place for discussion.

This is addressing #185 I think in good shape for early alpha review, but still has a few placeholders. Here is the link to the [notebook](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20access%20--%20Tutorial%20(part1).ipynb)

### CODE REVIEW

File: Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb [hunk 1]
```diff
@@ -0,0 +1,1401 @@
...
+    "                           native_filters=[(lambda x: x==tract_number, 'tract')])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask = np.zeros_like(m5map)\n",
```
Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: `m5map` has not yet been defined in this Notebook.

File: Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb [hunk 2]
```diff
@@ -0,0 +1,1401 @@
...
+    "hist2d(data_full['ra'], data_full['dec'],256); gca().set_aspect('equal'); \n",
+    "xlabel('RA [deg]');\n",
+    "ylabel('dec [deg]');\n",
+    "title('Full sample')\n",
+    "\n",
+    "subplot(122)\n",
+    "hist2d(data_cut['ra'], data_cut['dec'],256); gca().set_aspect('equal');\n",
+    "xlabel('RA [deg]');\n",
+    "ylabel('dec [deg]');\n",
+    "title('Clean objects');"
```
Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: Add a colorbar.

File: Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb [hunk 3]
```diff
@@ -0,0 +1,1401 @@
...
+      "text/plain": [
+       "<matplotlib.text.Text at 0x2ae54ada36d8>"
+      ]
+     },
+     "execution_count": 100,
+     "metadata": {},
+     "output_type": "execute_result"
+    },
+    {
+     "data": {
```
Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: Please remove plots.  If you have an image you want to persist, make a PNG, and include in assets.

File: Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb [hunk 4]
```diff
@@ -0,0 +1,1401 @@
...
...WmId0DuMsnTWKwDWKwmpUUh99nIcLT8PPFecz2/EyGVYeYxqWGsomo+QZkPy55WZIDJl4kI5OV2OzQcJNtNP9ikJWUrQaclWcv1dWO9QSeeFaFzrGLyYbr90NjnOlPqJiiS84ljMW3j+2o0AgO5x5T9dk9xPul6OTXL+nYY0TAm4fuotMUcIIYQQQlocPggSQgghhLQpNUvDpH4ypccMyVCoJiGLfGqVndNYUrQ1t1UuTtaoJcPeE1ECsKRXmafMDWwh0qJORGk5PSVHHxAdo/p4MqaarF7ZbwPi+Yl8rd23OjefbhdEMs3kHjSOZ80t+fq0bD6+O5FRNu6L56RL0E0Mdebmkc9i2siJqPuJ7Fn2WoCFvp5WPj8rz6MlwVbu01ST863vsXXOlsRc9IqAPueyVwkIaSZqzYs3c+hwrk3fxSSP4FyQjasSHLuda2MeweM3J9srT8fsAuIW7j0e790z5+ILP5/Z+iQA4DfOR9dwV3Afn/rpLWlb97nkvntudZR81z6frHGmxnMX+bwScRNTOq4OI4KEEEIIIW0KHwQJIYQQQtoUSsMNptbScDpZbplrWNotJ65GZDNrbi2pWeXiRP6r5hwtWkNWZszLgzLWkhg1ulTdzkeSeXQCYSFzPcN5WS5rS6LURCnaTios0qqWy63rIPPYcngcazmE+0bzjnAruXK/+m7Id0gn0ZY5+0tKGMq10e5pcT3XWv4QsGVZuV6ZVwWMz6Ao2TagvifG56yRY+vraTnshb4SdzEhzU5ZomQtj4qcPLs5tvkXksqFOnm0lHDLlG0zStAN/lPi8n3lQzGh9IbvnwMQHcUA0Hsk3sf3TW4HAKw+PJubb83B6Ngfvyq5l234/17P9bPOWSePrjXJ9lInmW6lpNaMCBJCCCGEtCmMCDaYMtOCFamQiFq1aI7Mo6N1VpRN5i7KJwjYpegkijhdLWoXoik66iVr0JEWiVJtVEM7jPOSSJHOmWeZJPQ5S7RHR6GiuaN6xAgAJjclLzhnTB6B/lxLghXBlKiezAcAvZvz5e0kOqav1+it8i/luC4xi+w6qj4zI4qWMfuECKU2tljRvDmjrJ5czw0vxijutFGezvqe6uiefGd1Hj7r+2dFZyWaqqOSurydZQCStencljv35A0fHQV5LLUxSUxImYjzK7npCGkpdKRMooPuWDRbXPj5G5N9x8+l
```
Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: Please remove plots.  If you have an image you want to persist, make a PNG, and include in assets.

Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: Remove output

Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: Remove output.

File: Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb [hunk 5]
```diff
@@ -0,0 +1,1401 @@
...
+    "Note: This notebook is intended to be run through the JupyterHub NERSC interface available here: https://jupyter-dev.nersc.gov. To setup your NERSC environment, please follow the instructions available here: https://confluence.slac.stanford.edu/display/LSSTDESC/Using+Jupyter-dev+at+NERSC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pylab inline"
```
Comment (wmwv) [inline Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb]: I'm very much a fan of being explicitly about `pyplot` and `numpy` use.  These tutorials also will end up being examples that people will copy for style.

I would suggest changing this one line to:
```
%matplotlib inline

import matplotlib.pyplot as plt
import numpy as np
```

and then encourage use of `np.` and `plt.` as neeeed below.

**Comment by EiffL** [comment]
@wmwv we have split the notebooks in 3, and I have renamed them as:
- [DC2 Coadd Run1.1p GCR tutorial -- Part I: GCR Access.ipynb](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20tutorial%20--%20Part%20I:%20GCR%20Access.ipynb)
- [DC2 Coadd Run1.1p GCR tutorial -- Part II: Lensing Cuts.ipynb](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20tutorial%20--%20Part%20II:%20Lensing%20Cuts.ipynb)
- [DC2 Coadd Run1.1p GCR tutorial -- Part III: Guided Challenges](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20tutorial%20--%20Part%20III:%20Guided%20Challenges.ipynb)

**Comment by yymao** [comment]
@EiffL how about updating the format of `native_filters` to `['tract==<tract_number>']` (lambda function still works, but string is easier to read).

**Comment by EiffL** [comment]
Sure, I'll do that and remove the last comments/placeholders we have

**Comment by wmwv** [comment]
Can you rename to not have -- or : in the filename, and potentially remove the spaces (I'm planning to go through later and rename the rest to not have spaces)?
: is an alternate path separator on Mac OSX.
-- is the "process no more options" indicator.

While obviously these all should be escaped, just like the spaces should be, in practice I think we will be better off to omit them.

You can still have them in the title of the Notebook, of course.

**Comment by EiffL** [comment]
Ok, thanks @wmwv I think notebooks I and II are in very good shape now, @yymao I've updated the code to use the new `native_filters` syntax, thanks for implementing that!

**Comment by EiffL** [comment]
Concerning Part III, which we intended as guided challenges, with incomplete cells that people would fill-in, how do you feel about them?

**Comment by EiffL** [comment]
Ok, rebased to master :-)

**Pull request merged**